umap/templates/umap/map_table.html

```diff
@@ -37,37 +37,84 @@
   </tbody>
+  <tfoot>
+    <tr>
+      <td colspan="5"></td>
+      <td>
+        <a href="{% url 'user_download' %}?{% spaceless %}
+        {% for map_inst in maps %}map_id={{ map_inst.pk }}{% if not forloop.last %}&{% endif %}{% endfor %}
+        {% endspaceless %}" class="button"
+          >{% blocktranslate count counter=maps.object_list|length %}
+            Download this map
+            {% plural %}
+            Download these {{ counter }} maps
+          {% endblocktranslate %}
+        </a>
+      </td>
+    </tr>
+  </tfoot>
 </table>
-<div class="pagination">
-  {% if maps.has_previous %}
-    <a href="?p=1{% if q %}&q={{ q }}{% endif %}">« {% translate "first" %}</a>
-    <a href="?p={{ maps.previous_page_number }}{% if q %}&q={{ q }}{% endif %}">‹ {% translate "previous" %}</a>
-  {% else %}
-    {# djlint:off #}
-    <span></span>
-    <span></span>
-    {# djlint:on #}
-  {% endif %}
+{% if maps.has_other_pages %}
+  <div class="pagination">
+    {% if maps.has_previous %}
+      <a href="?p=1{% if q %}&q={{ q }}{% endif %}">« {% translate "first" %}</a>
+      <a href="?p={{ maps.previous_page_number }}{% if q %}&q={{ q }}{% endif %}">‹ {% translate "previous" %}</a>
+    {% else %}
+      {# djlint:off #}
+      <span></span>
+      <span></span>
+      {# djlint:on #}
+    {% endif %}
 
-  <span class="current">
-    {% blocktranslate with maps_number=maps.number num_pages=maps.paginator.num_pages %}
-      Page {{ maps_number }} of {{ num_pages }}
-    {% endblocktranslate %}
-  </span>
+    <span class="current">
+      {% blocktranslate with maps_number=maps.number num_pages=maps.paginator.num_pages %}
+        Page {{ maps_number }} of {{ num_pages }}
+      {% endblocktranslate %}
+    </span>
 
-  {% if maps.has_next %}
-    <a href="?p={{ maps.next_page_number }}{% if q %}&q={{ q }}{% endif %}">{% translate "next" %} ›</a>
-    <a href="?p={{ maps.paginator.num_pages }}{% if q %}&q={{ q }}{% endif %}">{% translate "last" %} »</a>
-  {% else %}
-    {# djlint:off #}
-    <span></span>
-    <span></span>
-    {# djlint:on #}
-  {% endif %}
-</div>
+    {% if maps.has_next %}
+      <a href="?p={{ maps.next_page_number }}{% if q %}&q={{ q }}{% endif %}">{% translate "next" %} ›</a>
+      <a href="?p={{ maps.paginator.num_pages }}{% if q %}&q={{ q }}{% endif %}">{% translate "last" %} »</a>
+    {% else %}
+      {# djlint:off #}
+      <span></span>
+      <span></span>
+      {# djlint:on #}
+    {% endif %}
+  </div>
+{% endif %}
+<script type="text/javascript">
+  !(function () {
+    for (const deleteLink of document.querySelectorAll("table.maps a[href$='delete/']")) {
+      deleteLink.addEventListener('click', (event) => {
+        event.preventDefault()
+        const token = document.cookie.replace(
+          /(?:(?:^|.*;\s*)csrftoken\s*\=\s*([^;]*).*$)|^.*$/,
+          '$1'
+        )
+        if (confirm(L._('Are you sure you want to delete this map?'))) {
+          fetch(deleteLink.href, {
```

Comment:
TODO: use the new module from https://github.com/umap-project/umap/pull/1555/files